requirements.txt

```diff
@@ -1,2 +1,4 @@
 # List required packages in this file, one per line.
 h5py
+numpy
```

Comment:
h5py depends on numpy so technically redundant.

Reply:
Is there any downside to keeping it?

In C++, I picked up the habit of "if you use it; \#include it" and never depending on dependencies of dependencies being a given. In Python I generally follow the same policy for `requirements.txt`. The idea being to avoid future headaches as dependencies change locally or upstream.

Reply:
I don't see any downsides, and I support your approach. Imagine we move the hdf5-related code and the corresponding imports to a separate package, and the numpy dependency will disappear silently. So. +1 for listing it explicitly.

Reply:
Yeah, I follow the same approach: include anything that is directly imported.

Numpy is directly imported, not just used through h5py here. I suspect Tom mistakenly thought numpy was only a secondary dependency, because I’m pretty sure I learned this rule from him. :-)